inputremapper/gui/controller.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# input-remapper - GUI for device specific keyboard mappings
+# Copyright (C) 2022 sezanzeb <proxima@sezanzeb.de>
+#
+# This file is part of input-remapper.
+#
+# input-remapper is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# input-remapper is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with input-remapper.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations  # needed for the TYPE_CHECKING import
+
+import re
+from typing import TYPE_CHECKING, Optional, List, Tuple
+
+from gi.repository import Gtk, GLib
+
+from .data_bus import DataBus, MessageType, PresetData, StatusData, CombinationRecorded
+from .gettext import _
+from .data_manager import DataManager, DEFAULT_PRESET_NAME
+from .helper import is_helper_running
+from ..configs.mapping import MappingData, UIMapping
+from ..event_combination import EventCombination
+from ..injection.injector import RUNNING, FAILED, NO_GRAB, UPGRADE_EVDEV
+from ..input_event import InputEvent
+from ..logger import logger
+from .utils import CTX_SAVE, CTX_APPLY, CTX_KEYCODE, CTX_ERROR, CTX_WARNING, CTX_MAPPING
```

Comment:
What are your thoughts on relative vs absolute imports? I think other projects usually use absolute imports.

Reply:
Not sure. I was testing this out just to see if it is more readable. I don't really have a preference, other than that we should keep it consistent everywhere. 

What do you think?

Reply:
Relative imports have some special rules, absolute imports are easier to understand. So I'd go with absolute

Reply:
didn't know about that. Will change it to absolute.